### PR TITLE
feat!: Update AWS provider to support `v5.0` and increase Terraform MSV to `1.0`

### LINKED ISSUE #38: Ignore_headers_row changed to ignore_header_rows
Hi,

In main.tf line 269:
      ignore_headers_row                = lookup(s3_settings.value, "ignore_headers_row", null)

ignore_headers_row was changed to ignore_header_rows in https://registry.terraform.io/providers/hashicorp/aws/4.65.0/docs/resources/dms_endpoint

Thanks,
John

### LINKED ISSUE #43: Examples have stopped working
## Description
This is due to a provider change to 5.0 
 See
https://www.reddit.com/r/Terraform/comments/13uc728/an_issue_with_terraform_module_320/

Fix is to push the AWS provider version to 5.0, but this then brings a second issue with EBS volume size being removed from the Kafka provider. 

Recommend these be revisited to update the example to work with current providers, please.  

thanks. 

Should not be too difficult !

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Update AWS provider to support `v5.0` and increase Terraform MSV to `1.0`
- Add support for creating IAM access role
- Remove `s3_settings` from `aws_dms_endpoint` and use `aws_dms_s3_endpoint` instead per AWS provider direction
- Add support for `redshift_settings` on `aws_dms_endpoint`

## Motivation and Context
- Resolves #38
- Resolves #40
- Resolves #41 
- Resolves #43 

## Breaking Changes
- Yes

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
pending testing and validation + upgrade guidance

**Review by dannyibishev — COMMENTED** [review]
Looks like a good start. Theres also a deprecated argument that needs to be addressed.

`ignore_headers_row` is now deprecated. I think this needs to be replaced with `ignore_header_rows`

**Comment by brunodasilvalenga** [comment]
Hey @bryantbiggs any changes to have it approved? I'm looking to the fix of "ignore_header_rows".

File: examples/complete/main.tf [hunk 1]
```diff
@@ -2,12 +2,18 @@ provider "aws" {
...
   region = "us-east-1"
-  name   = "dms-ex-${replace(basename(path.cwd), "_", "-")}"
+  name   = "dms-ex-${basename(path.cwd)}"
+
+  vpc_cidr = "10.0.0.0/16"
+  azs      = slice(data.aws_availability_zones.available.names, 0, 3)
 
   db_name     = "example"
   db_username = "example"
+  db_password = "password123!" # do better!
```
Comment (antonbabenko) [inline examples/complete/main.tf:16]: https://www.youtube.com/watch?v=qztuEucrNBc 🤣

File: examples/complete/main.tf [hunk 2]
```diff
@@ -463,59 +412,18 @@ resource "aws_s3_object" "hr_data" {
...
-        },
-      ]
-    })
-  }
-
-  tags = local.tags
-}
-
 module "msk_cluster" {
   source  = "clowdhaus/msk-kafka-cluster/aws"
```
Comment (antonbabenko) [inline examples/complete/main.tf:416]:
```suggestion
  source  = "terraform-aws-modules/msk-kafka-cluster/aws"
```

File: versions.tf [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.17"
+      version = ">= 5.0"
     }
     time = {
       source  = "hashicorp/time"
-      version = ">=0.7.2"
+      version = ">=0.9"
```
Comment (antonbabenko) [inline versions.tf:11]:
```suggestion
      version = ">= 0.9"
```

**Review by antonbabenko — APPROVED** [review]
LGTM, just minor comments here and there.

File: versions.tf (hunk 3 above)
Comment (bryantbiggs) [reply versions.tf]: updated in https://github.com/terraform-aws-modules/terraform-aws-dms/pull/42/commits/6ef37ea00686d51687526f60f6bb9765bea7ef18

File: examples/complete/main.tf (hunk 2 above)
Comment (bryantbiggs) [reply examples/complete/main.tf]: updated in https://github.com/terraform-aws-modules/terraform-aws-dms/pull/42/commits/6ef37ea00686d51687526f60f6bb9765bea7ef18

**Comment by antonbabenko** [comment]
This PR is included in version 2.0.0 :tada:

**Pull request merged**